### PR TITLE
nsc-events-nextjs_5_262_modified-the-home-page-(app/page.tsx)-to-include-meta-tags

### PR DESCRIPTION
**Pull request by gitbiruk2010**
This is a pull request!

Please use the following format for the title:
< https://github.com/SeattleColleges/nsc-events-nextjs/tree/main > _ < sprint 5 Planning> _ < #262  > _ < Meta Data >

Please fill out this description with the following:
Resolves <issue #262 >
Description of what this pull request is for: Adding Meta Tags, Meta Images, and SEO to the NSC Event Next.js Home page app/page.tsx

Screenshots:
image

image

Enable auto-merge (squash) at the bottom of the page.

### CODE REVIEW

**Comment by nwm516** [comment]
It looks like your meta data is within the appropriate head element. There seems to be no spelling errors, missing parentheses, or closing tags. LGTM.

**Comment by miqo1385** [comment]
still learning a bit about this, but it looks good to me.

**Review by taylorpapke — COMMENTED** [review]
I'm not familiar with how to test this. Should I be able to check the page source and view these tags when I run your branch?

**Review by brinkbrink — APPROVED** [review]
I see that the meta tags have been added. I would like the product owner @heosman to sign off on the meta tags, unless you got this information from the product owner beforehand.

I also want to point out that there is a lot more to SEO that I would like you to research and implement. Take a look through Google's [Search Engine Optimization (SEO) Starter Guide](https://developers.google.com/search/docs/fundamentals/seo-starter-guide) and checklists like [this](https://backlinko.com/seo-checklist) to start. 

I have modified the issue and user story related to this pull request to reflect its incomplete status. Please reach out to me on Slack or respond to my comment here if you have any questions.

**Review by brinkbrink — DISMISSED** [review]
@heosman Can you please review these tags?

**Comment by brinkbrink** [comment]
I see that the meta tags have been added. I would like the product owner @heosman to sign off on the meta tags, unless you got this information from the product owner beforehand.

I also want to point out that there is a lot more to SEO that I would like you to research and implement. Take a look through Google's [Search Engine Optimization (SEO) Starter Guide](https://developers.google.com/search/docs/fundamentals/seo-starter-guide) and checklists like [this](https://backlinko.com/seo-checklist) to start. 

I have modified the issue and user story related to this pull request to reflect its incomplete status. Please reach out to me on Slack or respond to my comment here if you have any questions.

**Comment by brinkbrink** [comment]
> I'm not familiar with how to test this. Should I be able to check the page source and view these tags when I run your branch?

Hey @taylorpapke you can only view these tags in the file itself. I guess you could test the branch by making sure the tags don't disrupt any functionality but I feel that this is pretty straightforward and probably doesn't need that.

**Review by taylorpapke — APPROVED** [review]

**Comment by gitbiruk2010** [comment]
I am working on the SEO and I will do a PR on that as well.

**Comment by brinkbrink** [comment]
Thanks, @gitbiruk2010 !

**Review by heosman — CHANGES_REQUESTED** [review]
@gitbiruk2010 I can see the changes made in the home page file when I pull the branch on my IDE, but when I run the app, my browser title still says "Create Next App". Should it not be titled "North Seattle College Events" now? I don't see this title or those meta tags in the head when I use dev tools. It seems that the title and meta tags might be overridden by the metadata variable in layout.tsx.

@taylorpapke @brinkbrink Would we not be able to test the PR this way?

**Comment by gitbiruk2010** [comment]
Haven't checked if the changes I made are merged to the main. If it is, you
should see the tags added. You might wanna check if your pr is  the latest.
The changes made doesn't affect the the behavior of the web parts so it may
be another issue. Let me know what you find.

**Comment by brinkbrink** [comment]
> @gitbiruk2010 I can see the changes made in the home page file when I pull the branch on my IDE, but when I run the app, my browser title still says "Create Next App". Should it not be titled "North Seattle College Events" now? I don't see this title or those meta tags in the head when I use dev tools. It seems that the title and meta tags might be overridden by the metadata variable in layout.tsx.
> 
> @taylorpapke @brinkbrink Would we not be able to test the PR this way?

Yep, that makes sense.

**Comment by brinkbrink** [comment]
> @gitbiruk2010 I can see the changes made in the home page file when I pull the branch on my IDE, but when I run the app, my browser title still says "Create Next App". Should it not be titled "North Seattle College Events" now? I don't see this title or those meta tags in the head when I use dev tools. It seems that the title and meta tags might be overridden by the metadata variable in layout.tsx.
> 
> @taylorpapke @brinkbrink Would we not be able to test the PR this way?

No, not really, because the "Create React App" title is actually an issue with the HTML <title> tag. What is edited here is the metadata for a title, which is what web crawlers (what search engines use) will pick up on. We should definitely edit that <title> tag, though.

**Comment by brinkbrink** [comment]
But now that I'm looking closer, I see that this tag was also edited. When I run this branch locally neither the meta tags or title are changed:

Screenshot 2024-04-24 at 9 59 14 AM

Some suggestions:

1. "use client" could be interfering: https://www.reddit.com/r/nextjs/comments/11fjnpq/new_metadata_support_and_use_client/
2. We should be creating a Head component and then importing that: https://nextjs.org/docs/app/building-your-application/optimizing/metadata
3. We could also use a metadata object: https://nextjs.org/docs/app/building-your-application/optimizing/metadata

Thanks for the reminder that we should always pull the branch and test locally. @heosman @taylorpapke

**Comment by tinpham5614** [comment]
I have a question:
Should metadata be implemented in the layout.tsx as in the screenshot here? 
image

Resources:
https://nextjs.org/learn/dashboard-app/adding-metadata#page-title-and-descriptions

**Comment by brinkbrink** [comment]
> I have a question: Should metadata be implemented in the layout.tsx as in the screenshot here? image
> 
> Resources: https://nextjs.org/learn/dashboard-app/adding-metadata#page-title-and-descriptions

Yes, @tinpham5614, thank you for finding the metadata object. This is definitely the biggest issue here. The metadata object should be edited. For more information, @gitbiruk2010, reference the link in my previous comment.

**Comment by gitbiruk2010** [comment]
> I have a question: Should metadata be implemented in the layout.tsx as in the screenshot here? image
> 
> Resources: https://nextjs.org/learn/dashboard-app/adding-metadata#page-title-and-descriptions

You are Correct! i had the update implemented on page.tsx. I have made a correction and i will do a new pr. Thanks for catching that!

**Comment by gitbiruk2010** [comment]
image

**Comment by Diego-Cano** [comment]
Nice one on the SEO front for the app/page.tsx! Always a fan of seeing those meta elements getting improved. The screenshots are looking sharp, too!

Once we get this live, I’m keen to see how it’ll play out with our traffic stats. For anyone keen to brush up on SEO, Moz has some solid articles, and Search Engine Journal is like an encyclopedia for this stuff. 

Here are a couple of useful links:
https://www.searchenginejournal.com/
https://moz.com/blog/category/search-engines

**Review by brinkbrink — COMMENTED** [review]
Thank you for moving this to layout.tsx but it looks like the metadata object was not edited like I recommended above. It also appears that the meta properties in the head tag are pulling attributes from the metadata object, but those attributes do not exist. 

So when I test this branch locally there is no title and I can't find the metadata when inspecting the page:

Screenshot 2024-04-25 at 9 37 01 AM

I just edited this branch locally to show you what I'm saying. I deleted the head tag and edited the metadata object:

Screenshot 2024-04-25 at 9 32 52 AM

And now you can see a title for the page at the top of the browser window and the metadata I edited/added when inspecting the page.

Screenshot 2024-04-25 at 9 29 12 AM

Hope this clears things up.

**Review by brinkbrink — CHANGES_REQUESTED** [review]
Thank you for moving this to layout.tsx but it looks like the metadata object was not edited like I recommended above. It also appears that the meta properties in the head tag are pulling attributes from the metadata object, but those attributes do not exist. 

So when I test this branch locally there is no title and I can't find the metadata when inspecting the page:

Screenshot 2024-04-25 at 9 37 01 AM

I just edited this branch locally to show you what I'm saying. I deleted the head tag and edited the metadata object:

Screenshot 2024-04-25 at 9 32 52 AM

And now you can see a title for the page at the top of the browser window and the metadata I edited/added when inspecting the page.

Screenshot 2024-04-25 at 9 29 12 AM

Hope this clears things up.

File: app/page.tsx
Comment (brinkbrink) [inline app/page.tsx:1]: This should no longer be edited if we are using layout.tsx

**Comment by gitbiruk2010** [comment]
image

**Comment by gitbiruk2010** [comment]
NSC Events

**Comment by taylorpapke** [comment]
@brinkbrink @heosman Can you both please re-review this, see if the requested changes have been made and approve if they have?

@gitbiruk2010 You will also have to resolve the merge conflicts on this PR.

**Review by brinkbrink — CHANGES_REQUESTED** [review]
Screenshot 2024-05-01 at 9 25 36 AM

I see the title is working. That's great! But the rest of the metadata seems to still be incorrect. From my research on this issue I do not see a need for any metadata living in the <Head> tag because we have a metadata object doing this already. 

https://nextjs.org/learn/dashboard-app/adding-metadata
https://nextjs.org/docs/app/building-your-application/optimizing/metadata

**Review by heosman — COMMENTED** [review]
@gitbiruk2010 Like @brinkbrink mentioned, I can now see the title properly but the other metadata tags, such as the description, are still not visible to me in the browser.

Similar to the title metadata tag in this screenshot below, the other metadata tags you added, such as the description, should also be replacing the current metadata tags in the metadata object in the layout.tsx file.
image

Because of the existing metadata object in layout.tsx, the head element in the page.tsx file (in the app directory) is redundant, and can be removed.

**Review by JoyZhang2023 — CHANGES_REQUESTED** [review]
Looks good to me. All meta tags are in correct format with proper module imported. 
One thing may need revisit is that the page.tsx inherited the meta tag from layout.tsx. Thus the home page title is showing "NSC Events" as noted in layout.tsx instead of "North Seattle College Events" in page.tsx. 
image

**Comment by taylorpapke** [comment]
@gitbiruk2010 @brinkbrink @JoyZhang2023 What is the status for this PR? Is it reasonable to expect that we will be able to make the necessary changes, approve and resolve the conflicts here? If not, what course of action should we take?

**Comment by miqo1385** [comment]
the format seems correct, looks good to me

**Comment by kaimanasse15** [comment]
The Title looks good but I wonderd if anything is wrong with the database?
image

**Comment by taylorpapke** [comment]
Closing due to inactivity. Reopen if needed.

**Pull request closed without merging**